### PR TITLE
Make pylint happy

### PR DESCRIPTION
**Pull request by marcmengel**
- Open calls done with "with" and encoding=
- f-strings instead of %s stuff throughout
- misc ... return ... else: reworks
- some things flagged out  with #pylint: comments
- long strings broken down all over.

### CODE REVIEW

**Review by shreyb — CHANGES_REQUESTED** [review]
Just a couple of small concerns. I just want to make sure some of the huge changes have been tested. If so, this is good to merge.

File: lib/dagnabbit.py [hunk 1]
```diff
@@ -35,102 +38,102 @@ def parse_dagnabbit(
     in the dest directory, using global cmdline options from values
     along with ones parsed from dagnabbit file
     """
+    # pylint: disable=too-many-locals,too-many-branches,too-many-statements
     jinja_env = jinja.Environment(loader=jinja.FileSystemLoader(srcdir))
     jinja_env.filters["basename"] = os.path.basename
     proxy, token = creds.get_creds(values)
     count = 0
     linenum = 0
-    df = open(values["dag"], "r")
```
[the remark this answers is not in the document]
Comment (shreyb) [reply lib/dagnabbit.py:43]: Just a follow up - for both this and the comment in `condor_submit`, if you know that the pylint changes don't change the behavior, please consider these comments "resolved".

File: bin/condor_submit_dag [hunk 2]
```diff
@@ -17,72 +17,53 @@
...
+    os.environ["BEARER_TOKEN_FILE"] = token
 
+    if os.environ.get("GROUP", None) is None:
+        raise AttributeError(
+            f"{sys.argv[0]} needs -G group or $GROUP in the environment."
+        )
 
     schedd_add = get_schedd({})
-    schedd_name = schedd_add.eval("Machine")
+    # schedd_name = schedd_add.eval("Machine")
```
[the remark this answers is not in the document]
Comment (marcmengel) [reply bin/condor_submit_dag:63]: Sure, we could.  pylint says we don't use schedd_name anywhere...

[the remark this answers is not in the document]
Comment (marcmengel) [reply bin/condor_submit_dag:63]: Although part of me wants to leave it as an example of using the add to find out stuff...

[the remark this answers is not in the document]
Comment (marcmengel) [reply bin/condor_submit_dag]: deleted.

**Comment by marcmengel** [comment]
Just commenting that all the unit tests pass with these changes.

**Pull request merged**